### PR TITLE
Restore ClusterState version during remote state restore

### PR DESCRIPTION
**Pull request by linuxpi**
### Description
- During QL remote state restore we do not restore previous cluster state version
- With this change, we intent to restore the cluster state version back.

### Related Issues
Resolves https://github.com/opensearch-project/OpenSearch/issues/10692

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [ ] ~New functionality has javadoc added~
- [ ] ~Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))~
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))
- [ ] ~Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by linuxpi** [comment]
Flaky Test

[org.opensearch.repositories.azure.AzureBlobContainerRetriesTests.testWriteLargeBlob](https://build.ci.opensearch.org/job/gradle-check/28948/testReport/junit/org.opensearch.repositories.azure/AzureBlobContainerRetriesTests/testWriteLargeBlob/)  https://github.com/opensearch-project/OpenSearch/issues/6090
[org.opensearch.cluster.service.MasterServiceTests.testClusterStateBatchedUpdates](https://build.ci.opensearch.org/job/gradle-check/28948/testReport/junit/org.opensearch.cluster.service/MasterServiceTests/testClusterStateBatchedUpdates/) https://github.com/opensearch-project/OpenSearch/issues/10899

File: server/src/internalClusterTest/java/org/opensearch/remotestore/RemoteStoreClusterStateRestoreIT.java [hunk 1]
```diff
@@ -85,16 +85,19 @@ public void testFullClusterRestore() throws Exception {
...
         String prevClusterUUID = clusterService().state().metadata().clusterUUID();
+        long prevClusterVersion = clusterService().state().version();
 
         // Step - 2 Replace all nodes in the cluster with new nodes. This ensures new cluster state doesn't have previous index metadata
         resetCluster(dataNodeCount, clusterManagerNodeCount);
 
         String newClusterUUID = clusterService().state().metadata().clusterUUID();
         assert !Objects.equals(newClusterUUID, prevClusterUUID) : "cluster restart not successful. cluster uuid is same";
 
+        assert prevClusterVersion < clusterService().state().version() : "ClusterState version is not restored";
```
Comment (BhumikaSaini-Amazon) [inline server/src/internalClusterTest/java/org/opensearch/remotestore/RemoteStoreClusterStateRestoreIT.java:96]: nit : should we also report the `prevClusterVersion` and `clusterService().state().version()` values in the message?

Same for all next occurences

**Review by BhumikaSaini-Amazon — COMMENTED** [review]
Do we need to add assertions to the existing tests to validate that the ClusterState version is indeed restored?

File: server/src/test/java/org/opensearch/gateway/remote/RemoteClusterStateServiceTests.java [hunk 2]
```diff
@@ -750,10 +753,10 @@ public void testReadGlobalMetadata() throws IOException {
...
         mockBlobContainerForGlobalMetadata(mockBlobStoreObjects(), expectedManifest, expactedMetadata);
 
-        Metadata metadata = remoteClusterStateService.getLatestMetadata(
+        Metadata metadata = remoteClusterStateService.getLatestClusterState(
             clusterState.getClusterName().value(),
             clusterState.metadata().clusterUUID()
-        );
+        ).getMetadata();
 
         assertTrue(Metadata.isGlobalStateEquals(metadata, expactedMetadata));
```
Comment (shwetathareja) [inline server/src/test/java/org/opensearch/gateway/remote/RemoteClusterStateServiceTests.java:761]: assert on cluster state version as well

**Review by shwetathareja — APPROVED** [review]

File: server/src/internalClusterTest/java/org/opensearch/remotestore/RemoteStoreClusterStateRestoreIT.java (hunk 1 above)
Comment (linuxpi) [reply server/src/internalClusterTest/java/org/opensearch/remotestore/RemoteStoreClusterStateRestoreIT.java:96]: yes, we can add that. Thanks @BhumikaSaini-Amazon

File: server/src/test/java/org/opensearch/gateway/remote/RemoteClusterStateServiceTests.java (hunk 2 above)
Comment (linuxpi) [reply server/src/test/java/org/opensearch/gateway/remote/RemoteClusterStateServiceTests.java]: Sure, added that. Thanks @shwetathareja

**Comment by linuxpi** [comment]
> Do we need to add assertions to the existing tests to validate that the ClusterState version is indeed restored?

We have added assertions to validate new cluster state version would always be greater than previous version

**Review by shwetathareja — APPROVED** [review]

**Review by BhumikaSaini-Amazon — APPROVED** [review]
Thanks @linuxpi!

**Comment by linuxpi** [comment]
will raise the backport pr manually

**Pull request merged**